functions/github.ts

```diff
@@ -0,0 +1,32 @@
+import * as firebaseAdmin from 'firebase-admin';
+import * as firebaseFunctions from 'firebase-functions';
+import {setGithubStatus} from './util/github';
+
+/** Github status update token */
+const token = firebaseFunctions.config().secret.github;
+
+/** The repo slug. This is used to validate the JWT is sent from correct repo. */
+const repoSlug = firebaseFunctions.config().repo.slug;
+
+/** Domain to view the screenshots */
+const authDomain = firebaseFunctions.config().firebase.authDomain;
+
+/** The same of this screenshot testing tool */
+const toolName = firebaseFunctions.config().tool.name;
+
+export function updateGithubStatus(event: firebaseFunctions.Event<any>) {
+  if (!event.data.exists() || typeof event.data.val() != 'boolean') {
+    return;
+  }
+  let result = event.data.val() == true;
+  let prNumber = event.params.prNumber;
+  return event.data.ref.parent.child('sha').once('value').then((sha: firebaseAdmin.database.DataSnapshot) => {
+    return setGithubStatus(sha.val(),
```

Comment:
I think the API we used in `/tools/gulp/util/github` is a bit more convenient. You just pass in a JSON config instead of X parameters.